### DISCUSSION TITLE
Newbie question

### DISCUSSION THREAD

**Discussion by MSDaytona** [MiguelRipoll23/homebridge-securitysystem#370] [Q&A]
I have successfully installed the system on my hoobs box and set-up some automation in HomeKit to switch to the away, home etc. mode depending. It works fine.

Now the question:
What is the relation between the different modes ( Off, Away, Home, Night etc. ) and the Siren (  called "Trip" in my HomeKit ).
I do not get it.  Please enlighten me.

**Comment by MiguelRipoll23** [comment]
Hello,
The trip switches are necessary to trigger the security system, in order to create easier automations (condition-less) additional trip switches are provided for each mode that will only work if the mode is set.
If the mode isn't set it will return an error and the switch will remain off.

**Comment by MSDaytona** [comment]
Hi Miguel

thanks for coming back to me.


I still don’t get it.


I have so far set-up two automations:

1.) When I leave home set the mode to away   and
2.) When I come home set the mode to home

Works o.k.

If I understand you correctly this would be the prerequisite for for the trip switches. Right ?

The trip switches I would need to incorporate into another automation say
- if motion detected then trigger trip switch away which would then trigger whatever I want

But the trip switch away would only work if the mode switch is away as well, else nothing would happen. Right ?


Appreciate your assistance

Michael


(:-)--------------------------------------------------(-:)
     In a world without walls and fences,
         who needs windows and gates.

 Michael Schulz        Sidcup/Kent/England
           ***@***.***
               www.michael-schulz.net
(:-)--------------------------------------------------(-:)


￼
Help preserve the colour of our world – Think before you print.

**Reply by MiguelRipoll23** [reply]
That’s right

**Comment by MSDaytona** [comment]
Thank you.

Got it now.


(:-)--------------------------------------------------(-:)
     In a world without walls and fences,
         who needs windows and gates.

 Michael Schulz        Sidcup/Kent/England
           ***@***.***
               www.michael-schulz.net
(:-)--------------------------------------------------(-:)


￼
Help preserve the colour of our world – Think before you print.

**Comment by MSDaytona** [comment]
Hi Miguel,

I am very sorry but I am still struggling.  Please be so kind and put me on the right track.

1.) Security system is successfully installed
2.) In the Mac HomeApp I can see the different mode switches ( Away, Home, Night, Off and Pause )
3.) I configured the plugin so that I only have one Siren Swich ( in my case called “Trip” )
4.) Depending which status ( Away, Home, etc. ) I select the corresponding Mode switch is changing from Off to On
     The siren switch ( TRIP ) is still showing OFF at this point.

As clarified in our correspondence yesterday the Mode switch “ON” is a prerequisite.

But now I am struggling on how for example I bring the Siren switch into the logic that when for example
Certain windows or doors are Not closed the alarm goes off.

I started setting up an automation that when the accessory state of the windows/doors is OPEN and then I got stuck.

Your guidance would be very much appreciated 

Thanks in advance


Michael


(:-)--------------------------------------------------(-:)
     In a world without walls and fences,
         who needs windows and gates.

 Michael Schulz        Sidcup/Kent/England
           ***@***.***
               www.michael-schulz.net
(:-)--------------------------------------------------(-:)


￼	
Help preserve the colour of our world – Think before you print.

**Reply by MiguelRipoll23** [reply]
No worries. So you configured the plugin to show the mode switches and the global trip switch, you set a mode using the mode switches and the mode switch turned on.. as long as the mode you set isn’t “Off” if you turn on the trip switch it it will turn on and remain on. At this point the security system will trigger immediately or after a few seconds depending of your configuration.

**Reply by MiguelRipoll23** [reply]
Using the global trip switch means that any mode will trip the security system. If not clear yet, please provide a screenshot or the plugin log.

**Comment by MSDaytona** [comment]
Very sorry

but I am still not sure I get it.

Does what you tell me mean that when I set the mode switch to ON I have to switch the TRIP switch to ON as well ?

I just checked but if switch the trip switch to On it doesn’t stay ON 


The plug-in config is:
  "accessories": [
        {
            "name": "Security System",
            "default_mode": "Off",
            "arm_seconds": 30,
            "trigger_seconds": 0,
            "reset_minutes": 10,
            "home_arm_seconds": "0",
            "night_arm_seconds": "0",
            "away_arm_seconds": "30",
            "home_trigger_seconds": "null",
            "night_trigger_seconds": "null",
            "away_trigger_seconds": "null",
            "away_extended_trigger_seconds": "null",
            "double_knock_seconds": 90,
            "home_double_knock_seconds": "null",
            "away_double_knock_seconds": "null",
            "night_double_knock_seconds": "null",
            "tripped_sensor_seconds": 5,
            "siren_sensor_seconds": 5,
            "pause_minutes": 0,
            "audio_language": "en-US",
            "audio_volume": 100,
            "accessory": "security-system",
            "plugin_map": {
                "plugin_name": "homebridge-securitysystem"
            },
            "mode_off_switch": true,
            "siren_override_switch": true,
            "mode_switches": true,
            "siren_mode_switches": false,
            "mode_pause_switch": true,
            "mode_away_extended_switch": false,
            "siren_switch": true
        }
    ]
}

(:-)--------------------------------------------------(-:)
     In a world without walls and fences,
         who needs windows and gates.

 Michael Schulz        Sidcup/Kent/England
           ***@***.***
               www.michael-schulz.net
(:-)--------------------------------------------------(-:)


￼	
Help preserve the colour of our world – Think before you print.

**Reply by MiguelRipoll23** [reply]
Let’s say that for example you turn on the “Mode Away” switch to set the Away mode and the mode changes correctly. Now if you have an automation that if motion is detected turns on the “Trip” switch if the automation runs the security system will trigger either immediately or after a few seconds

**Comment by MSDaytona** [comment]
You are very helpful.

So, if I understand correctly:

I would need to set-up an automation like

      if motion detected or window  or door opened then set TRIP switch to ON ( at this point in time the mode switch would already be ON )

but where do I for example automate the alarm which should now follow ?


(:-)--------------------------------------------------(-:)
     In a world without walls and fences,
         who needs windows and gates.

 Michael Schulz        Sidcup/Kent/England
           ***@***.***
               www.michael-schulz.net
(:-)--------------------------------------------------(-:)


￼
Help preserve the colour of our world – Think before you print.

**Reply by MiguelRipoll23** [reply]
I’m not sure if you mean how to create the automations or how to automate the modes when you enter or leave your home. If you want to create an automation it should be possible inside the Home app (bottom menu) and if you want to automate the modes set the same area but use the mode switches to actually be able to run the automations using the someone entera / leave the house as triggers.

**Comment by MSDaytona** [comment]
Hi again,

I think I didn’t make myself clear:

1.) When switching for example the Alarm system manually from “OFF” to “AWAY”  ‘ll do that either manually  or use the “People Leave” function of the Mac Home app.  No problem, this already works
     perfectly.

2.) Based on todays mails I have now learned that I have to set-up an automation which for example “if motion is detected then change the TRIP switch to “ON” . Very straightforward, no problem.


This means that when a motion is detected both the mode switch as well as the trip switch would be “ON”.    Until here I have now understood everything.


The question I am having now is that at that point ( both switches are “ON” ) something needs to happen like for example one of my HomePods should play something.

How do I do that ??

(:-)--------------------------------------------------(-:)
     In a world without walls and fences,
         who needs windows and gates.

 Michael Schulz        Sidcup/Kent/England
           ***@***.***
               www.michael-schulz.net
(:-)--------------------------------------------------(-:)


￼
Help preserve the colour of our world – Think before you print.

**Comment by MSDaytona** [comment]
Hi Miguel

forget my  last email.

I figured it out:

I set-up an automation that when the TRIP switch goes to ON an alarm siren on one of my HomePods goes on. Work now perfectly.

Apologies for the inconvenience

and thank you very much for your help



Michael


(:-)--------------------------------------------------(-:)
     In a world without walls and fences,
         who needs windows and gates.

 Michael Schulz        Sidcup/Kent/England
           ***@***.***
               www.michael-schulz.net
(:-)--------------------------------------------------(-:)


￼	
Help preserve the colour of our world – Think before you print.
